contracts/AssetPool.sol

```diff
@@ -1,41 +1,56 @@
 // SPDX-License-Identifier: MIT
-pragma solidity >=0.4.22 <0.9.0;
+pragma solidity ^0.8.0;
 import "./LiquidityPoolInterfaces.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "hardhat/console.sol";
+import "./Proxy.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
 
-contract AssetPool is LiquidityPoolInterfaces, Ownable, ReentrancyGuard, ERC20("Cruize Protection LP Token", "cr") {
+contract AssetPool is Proxy, Ownable, ReentrancyGuard {
   using SafeMath for uint256;
-  mapping(address => mapping(address => uint)) public userInfo;
+  mapping(address => mapping(address => uint256)) public userInfo;
```

Comment:
can remove. we can call the contract balance of function in to withdraw function.